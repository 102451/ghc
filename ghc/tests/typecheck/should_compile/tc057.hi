interface ShouldSucceed where {
class Eq' a where { deq }
instance <Eq' Int>
instance Eq' a => <Eq' [a]>
dand  ::  Bool -> Bool -> Bool
f  ::  Eq' t93 => t93 -> t93 -> Bool
}
