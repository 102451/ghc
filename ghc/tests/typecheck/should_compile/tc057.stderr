ghc: module version changed to 1; reason: no old .hi file
_exports_
ShouldSucceed dand f Eq'{:Eq' :Eq' deq};
_instances_
instance _forall_ [$a] {Eq' $a} => {Eq' [$a]} = $dEq'0;
instance {Eq' PrelBase.Int} = $dEq'Int0;
_declarations_
1 $dEq'0 _:_ _forall_ [$a] {Eq' $a} => {Eq' [$a]} ;;
1 $dEq'Int0 _:_ {Eq' PrelBase.Int} ;;
1 class Eq' $r3W where {deq :: $r3W -> $r3W -> PrelBase.Bool} ;
1 dand _:_ PrelBase.Bool -> PrelBase.Bool -> PrelBase.Bool ;;
1 f _:_ _forall_ [$a] {Eq' $a} => $a -> $a -> PrelBase.Bool ;;
